src/main/java/com/dataloom/neuron/Synapse.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2017. Kryptnostic, Inc (dba Loom)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * You can contact the owner of the copyright at support@thedataloom.com
+ */
+
+package com.dataloom.neuron;
+
+public interface Synapse<T extends Signal> {
```

Comment:
There is already a class with this name and I think Receptor void signal( AuditableEvent event ) seems like a better pattern here as it will standardized the envelope instead of requiring the use of generics (at this level).